### ISSUE TITLE
Runtime error with Webpack v5

### ISSUE THREAD

**Issue by benochr** [palantir/blueprint#4393]
#### Environment

- __Package version(s)__: 3.35.0
- __Operating System__: MacOS Catalina 10.15.7
- __Browser name and version__: Chrome 86.0.4240.111

#### Actual behavior

When building with Webpack v5.3.0 we get a runtime error.

```
classes.js:20 Uncaught ReferenceError: process is not defined
    at Module../node_modules/@blueprintjs/core/lib/esm/common/classes.js (classes.js:20)
    at __webpack_require__ (bootstrap:18)
```

Webpack 5 no longer includes polyfills for it, as described [here](https://webpack.js.org/migrate/5/#run-a-single-build-and-follow-advises)

#### Expected behavior

No runtime errors. Support builds with webpack 5.

**Comment by adidahiya** [comment]
I'll update the code to avoid referencing `process` directly. In the meantime you can work around this with `DefinePlugin` or `EnvironmentPlugin`.

**Comment by jasongrout** [comment]
This would also fix #3703 

> In the meantime you can work around this with `DefinePlugin` or `EnvironmentPlugin`.

See https://github.com/palantir/blueprint/issues/3739#issuecomment-602024694 for an example.

**Comment by adidahiya** [comment]
fixed by #4934
